internal/functions/serve/serve.go

```diff
@@ -59,179 +52,84 @@ func (i *RuntimeOption) toArgs() []string {
 	return flags
 }
 
-const (
-	dockerRuntimeServerPort    = 8081
-	dockerRuntimeInspectorPort = 8083
-)
-
-//go:embed templates/main.ts
-var mainFuncEmbed string
-
 func Run(ctx context.Context, envFilePath string, noVerifyJWT *bool, importMapPath string, runtimeOption RuntimeOption, fsys afero.Fs) error {
+	return RunWithWatcher(ctx, envFilePath, noVerifyJWT, importMapPath, runtimeOption, fsys, &RealFileWatcherSetup{})
```

Comment:
How about we only run the file watcher if user sets the `policy = "per_worker"` in edge runtime config?

In oneshot policy, there's no need to watch for file changes.

Reply:
Hmm in those cases we only need to update the `SUPABASE_INTERNAL_FUNCTIONS_CONFIG` env var. But yes file watching handles this use case too.